### ISSUE TITLE
[Feature]: Add warning for no doc updates 

### ISSUE THREAD

**Issue by phklive** [0xMiden/miden-node#587]
### To whom is this feature for?

App developer

### Why is is this feature needed?

To make sure that our docs are always up to date.

### How is this feature used?

Follow the update that we have made for the changelog and adapt it for the doc updates: https://github.com/0xPolygonMiden/miden-base/pull/799

Make sure that each PR has a corresponding doc update if needed. 

If not the PR should be flagged with `no docs`

**Comment by Mirko-von-Leipzig** [comment]
What sort of changes would require documentation updates?

**Comment by phklive** [comment]
> What sort of changes would require documentation updates?

Everything that touches to something that is documented. 

Ex. You have docs about the mempool, you change the mempool impl, you need to make sure to update the mempool docs

**Comment by Mirko-von-Leipzig** [comment]
> > What sort of changes would require documentation updates?
> 
> Everything that touches to something that is documented.
> 
> Ex. You have docs about the mempool, you change the mempool impl, you need to make sure to update the mempool docs

But there are no mempool docs unless you mean comments in the code

**Comment by bobbinth** [comment]
I think over time it would be good to have documentation that describes how mempool (and other components) work in a more of an "article" format and put these into an mdbook. For now, we don't really have this (and I'm not sure we need it for the immediate future).

But, I think we should have docs on how to start the node, monitor it (when this is implemented), maintain etc. We have some of this in Markdown files, but may make sense to transfer this to an mdBook format. Another thing that could go there is documentation on API endpoints (especially for the RPC component).

Basically, I think we should move most of the existing docs into the mdbook format, add RPC API docs to it, and then these docs would be subject to updates when we change things. The deeper architectural docs could come later.

**Comment by Himess** [comment]
@bobbinth , If it's ok , I can add some doc to miden-node like https://github.com/0xMiden/crypto/pull/408

**Comment by bobbinth** [comment]
Closing this as it seems like this creates more friction than benefits.
